lib/src/facts/posix/identity_resolver.cc

```diff
@@ -25,7 +25,15 @@ namespace facter { namespace facts { namespace posix {
         uid_t uid = geteuid();
         struct passwd pwd;
         struct passwd *pwd_ptr;
-        int err = getpwuid_r(uid, &pwd, buffer.data(), buffer.size(), &pwd_ptr);
+        int err = -1;
+
+        do {
+            err = getpwuid_r(uid, &pwd, buffer.data(), buffer.size(), &pwd_ptr);
+            if (err == ERANGE) {
+                buffer.resize(buffer.size() + 1024);
```

Comment:
I do agree that such large entries should be pretty rare.
Thanks for the review @branan !